apps/src/lib/kits/maker/ui/SetupGuide.jsx

```diff
@@ -54,32 +55,53 @@ export default class SetupGuide extends React.Component {
     );
     const {webSerialPort} = this.state;
 
-    // WebSerial requires user input for user to select port.
-    // Add a button for user interaction before initiated Setup Checklist
-    if (shouldUseWebSerial() && !webSerialPort) {
-      return (
-        <input
-          style={{marginLeft: 9, marginTop: -4}}
-          className="btn"
-          type="button"
-          value={'Connect to Board'}
-          onClick={() => {
-            navigator.serial
-              .requestPort({filters: WEB_SERIAL_FILTERS})
-              .then(port => {
-                this.setState({webSerialPort: port});
-              });
-          }}
-        />
-      );
+    let webSerialRender;
```

Comment:
Nit: I feel like the more common pattern I see is to extract this to a function like renderWebSerial(), and put this conditional logic there (rather than having it directly in the render function)? Maybe this is more standard in star labs though.

Reply:
Along the same lines, I appreciated the way that you handled ChromebookInstructions earlier with the webSerial and chromeApp functions  pulled out from render.